### PR TITLE
Upgrade to operator-sdk 1.0

### LINKED ISSUE #166: Upgrade to operator-sdk 1.0
There is serious amount of changes and cool stuff(tm) in 1.0 sdk release, most notably project structure change and full switch to kubebuilder as a generator engine under the hood. Also  kuttl and envtest  integration worth special attention

* https://www.openshift.com/blog/operator-sdk-reaches-v1.0
* https://github.com/operator-framework/operator-sdk/releases/tag/v1.0.0
* https://sdk.operatorframework.io/docs/building-operators/golang/project_migration_guide/#migration-steps
* https://sdk.operatorframework.io/docs/advanced-topics/scorecard/kuttl-tests/
* https://book.kubebuilder.io/reference/testing/envtest.html#using-envtest-in-integration-tests 
* https://sdk.operatorframework.io/docs/upgrading-sdk-version/

### PR DESCRIPTION
**Pull request by ytsarev**
Upgrade according to https://sdk.operatorframework.io/docs/building-operators/golang/project_migration_guide/ 

Outstanding issues that were solved
* Effectively it is completely new project structure and engine, the logic was cherry-picked and migrated step by step in multiple places
* DepResolver introduced some dependency hell, we included env.go from kuberentes utils right in the codebase
* Invalid memory nil pointer access was solved by adopting of depresolver to new style of GslbReconciler initialization
* Adapt to new reconciliation related method signatures
* Adapt to new logging style
* Upstream/k8gb Makefile merge
* Unit test fixes - controller runtime fake client changed its behaviour in treating object versions 
* Terratest fixes
* Github actions changes

Resolves #166 

All tests are passing.

### CODE REVIEW

File: controllers/gslb_controller_test.go [hunk 1]
```diff
@@ -650,7 +651,7 @@ func TestGslbController(t *testing.T) {
 			t.Fatalf("Can't setup env var: (%v)", err)
 		}
 		resolver := depresolver.NewDependencyResolver(context.TODO(), cl)
-		r.config, err = resolver.ResolveOperatorConfig()
+		r.Config, err = resolver.ResolveOperatorConfig()
```
Comment (kuritka) [inline controllers/gslb_controller_test.go:654]: I created [an additional issue](https://github.com/AbsaOSS/k8gb/issues/168) for that

File: go.mod [hunk 2]
```diff
@@ -3,47 +3,17 @@ module github.com/AbsaOSS/k8gb
 go 1.13
```
Comment (kuritka) [inline go.mod:3]: Can we try to migrate to go1.15? OR in separate issue ?

File: go.mod [hunk 3]
```diff
@@ -3,47 +3,17 @@ module github.com/AbsaOSS/k8gb
...
+	github.com/stretchr/testify v1.5.1
+	k8s.io/api v0.18.8
+	k8s.io/apimachinery v0.18.8
+	k8s.io/client-go v0.18.8
+	sigs.k8s.io/controller-runtime v0.6.2
+	sigs.k8s.io/external-dns v0.7.4
 )
-
-// Pinned to kubernetes-1.16.2
-replace (
```
Comment (kuritka) [inline go.mod:22]: I'm so happy we got rid of that! 🥇

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,28 @@
...
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Kubernetes Generated files - skip generated files, except for vendored files
+
+!vendor/**/zz_generated.*
+
+# editor and IDE paraphernalia
+.idea
+*.swp
```
Comment (kuritka) [inline .gitignore:22]: maybe we can add `.vscode` as well.

**Review by kuritka — CHANGES_REQUESTED** [review]
Roughly it is very good.

File: go.mod (hunk 2 above)
Comment (ytsarev) [reply go.mod:3]: @kuritka I would better not mix it , also `go 1.13` spec was created by framework generator so don't really want to override it given complexity of this PR

File: .gitignore (hunk 4 above)
Comment (ytsarev) [reply .gitignore:22]: @kuritka it's actually there https://github.com/AbsaOSS/k8gb/blob/6d05dc68d0c52d70b45e2971e57294b69270bc6f/.gitignore#L100 
.gitignore is a bit messy given the merge with upstream, i just sequentially combined them as of now

**Review by kuritka — APPROVED** [review]

**Review by somaritane — APPROVED** [review]
Great job, lgtm!

**Pull request merged**